### DISCUSSION TITLE
CompileProblem: A template name was expected instead of 'unbounded_value_sequence

### DISCUSSION THREAD

**Discussion by CF-Air** [DOCGroup/ACE_TAO#1805] [Q&A] [answered]
Hi everyone,
I've got a Problem building TAO on my Solaris-System. The make finished in ACE. I started in TAO and got stuck at PolicyManager.cpp.
It complains
../tao/Unbounded_Octet_Sequence_T.h", line 49: Error: A template name was expected instead of 'unbounded_value_sequence'.
"../tao/Unbounded_Octet_Sequence_T.h", line 50: Error: Type must be specified for "unbounded_value_sequence".
"../tao/Unbounded_Octet_Sequence_T.h", line 56: Error: A template name was expected instead of 'unbounded_value_sequence'.
"../tao/Unbounded_Octet_Sequence_T.h", line 57: Error: Type must be specified for "unbounded_value_sequence".
...

My command ist
CC -mt -g -library=stdcpp -m32 -std=c++14 -g -features=zla -mt -fast -library=stdcpp   -DACE_HAS_KSTAT -DACE_HAS_CUSTOM_EXPORT_MACROS=0 -D_POSIX_PTHREAD_SEMANTICS   -IXXX/ACE_wrappersSolarisCC -DACE_HAS_SCTP -DACE_HAS_LKSCTP -D__ACE_INLINE__ -I../.. -I.. -DTAO_HAS_VALUETYPE_OUT_INDIRECTION -DTAO_BUILD_DLL  -c -KPIC -o sun4/.shobj/Policy_Manager.o Policy_Manager.cpp

Has anyone an idea how to fix this problem?

Thanks in advance,   Carmen

SunOS 5.11 11.3 sun4v sparc sun4v, CC: Studio 12.6 Sun C++ 5.15 SunOS_sparc 2017/05/30

**Comment by jwillemsen** [comment, marked as answer]
TAO on solaris version hasn't been maintained for years, it could be that the configuration file needs to be updated or you maybe hit a compiler bug. 

You don't mention which TAO version you are using, maybe an older release could work. Also you enable C++14 support, probably that hasn't been tried yet on Solaris at all.

Be aware that github support is best effort, when you need guaranteed response, see https://github.com/DOCGroup/ACE_TAO/wiki/ACE-and-TAO-Commercial-support for the support options, including @RemedyIT, the company I work for.

**Comment by CF-Air** [comment]
Thanks, at the moment I try to compile 2.2a with patchlevel 21. Some years ago I compiled 2.2a patchlevel 13, this was ok for the same compilerflags.
We've got some communication problems, not reproducable, so I was thinking about an upgrade.
So I will try the commercial support.

**Reply by jwillemsen** [reply]
The 2.2a release is already pretty old, have you tried one of the latest TAO releases, see https://github.com/DOCGroup/ACE_TAO/releases

**Reply by CF-Air** [reply]
no, I'll have a try; tanks
